src/js/tech/html5.js

```diff
@@ -203,6 +207,46 @@ class Html5 extends Tech {
     });
   }
 
+  /**
+   * Handle IOS Headphone disconnection during playback
+   *
+   * @private
+  */
+  handleIOSHeadphonesDisconnection_() {
+    // Fudge factor to account for TimeRanges rounding
+    const TIME_FUDGE_FACTOR = 1 / 30;
+
+    // Comparisons between time values such as current time and the end of the buffered range
+    // can be misleading because of precision differences or when the current media has poorly
+    // aligned audio and video, which can cause values to be slightly off from what you would
+    // expect. This value is what we consider to be safe to use in such comparisons to account
+    // for these scenarios.
+    const SAFE_TIME_DELTA = TIME_FUDGE_FACTOR * 3;
+
+    // If iOS check if we have a real stalled or supend event or
+    // we got stalled/suspend due headphones where disconnected during playback
+    this.on(['stalled', 'suspend'], (e) => {
+      const buffered = this.buffered();
+
+      if (buffered.length > 0) {
+        let extraBuffer = false;
+
+        // Establish if we have an extra buffer in the current time range playing.
+        for (let i = 0; i < buffered.length; i++) {
+          if (buffered.start(i) <= this.currentTime() < buffered.end(i) + SAFE_TIME_DELTA) {
+            extraBuffer = true;
+            break;
+          }
+        }
+
+        // if tech is not paused, browser has internet connection & player has extraBuffer inside the timeRange
+        if ((!this.paused() && window.navigator.onLine) && extraBuffer) {
```

Comment:
I don't think the extra parens are necessary as the expression will just evaluate each item in turn.
```suggestion
        if (!this.paused() && window.navigator.onLine && extraBuffer) {
```

Reply:
Also, should we check extraBuffer first?

Reply:
Make sense and we can decide quicker the state of the playback.